library/alloc/src/vec.rs

```diff
@@ -243,8 +243,8 @@ use crate::raw_vec::RawVec;
 /// * It would penalize the general case, incurring an additional branch
 ///   on every access.
 ///
-/// `Vec` will never automatically shrink itself, even if completely empty. This
-/// ensures no unnecessary allocations or deallocations occur. Emptying a `Vec`
+/// `Vec` will not automatically shrink itself, even if completely empty, when doing so
+/// would cause unnecessary allocations or deallocations to occur. Emptying a `Vec`
```

Comment:
I believe playing with words would confuse most users, this does not feel much different to me. Maybe
```suggestion
/// `Vec` will not automatically shrink itself, even if completely empty. **But** it
/// will try to reduce unnecessary allocations or deallocations by reusing existing allocation. Emptying a `Vec`
```
Maybe a big **but** helps? Probably needs rewrapping.